consensus/propagation/commitment_state.go

```diff
@@ -10,21 +10,21 @@ import (
 
 type proposalData struct {
 	compactBlock *proptypes.CompactBlock
-	block        *types.PartSet
+	block        *proptypes.CombinedPartSet
 	maxRequests  *bits.BitArray
 }
 
 type ProposalCache struct {
 	store         *store.BlockStore
-	pmtx          *sync.RWMutex
+	pmtx          *sync.Mutex
```

Comment:
the rw mutex adds complexity and likely actually makes performance worse since reads are so fast